core/src/main/scala/org/apache/gearpump/cluster/client/ClientContext.scala

```diff
@@ -49,9 +48,12 @@ class ClientContext(config: Config, sys:Option[ActorSystem], _master: Option[Act
   implicit val system = sys.getOrElse(ActorSystem(s"client${Util.randInt}" , config))
   LOG.info(s"Starting system ${system.name}")
 
-  import system.dispatcher
-
   private val master = _master.getOrElse(system.actorOf(MasterProxy.props(masters), s"masterproxy${system.name}"))
+  private val jarStoreService = JarStoreService.get(config)
+  jarStoreService match {
+    case needConfig: ConfigRequired => needConfig.init(config)
+    case needSystem: ActorSystemRequired => needSystem.init(system)
+  }
```

Comment:
What if a  service require both ActorSystem and Config?

case match will only do a partial match.